src/components/AddItemDialog.vue

```diff
@@ -98,68 +103,113 @@ function getRootItemPath(): string {
+function editItem(newItem: IItem): void {
+  treeStore.editItem({
+    ...newItem,
+    key: props.currentItem.key,
+    to: props.currentItem.to,
+  });
+  hideDialog();
+}
+
+function modifyItem(newItem) {
+  newItem = {
+    ...newItem,
+    quantity: addForm.quantity,
+    tags: addForm.tags,
+    description: addForm.description,
+  };
+
+  if (!formErrors.errorLabel) {
+    if (props.currentItem) {
+      editItem(newItem);
+    } else {
+      addTreeNode(newItem, "/item/");
+    }
+  }
+}
+
+function modifySection(newItem: IItem) {
+  newItem = {
+    ...newItem,
+    items: [],
+  };
+  formErrors.errorQuantity = validateQuantity(addForm.quantity); //quantity -> count
+  formErrors.errorDescription = validateName(addForm.description, 264);
+  if (
+    !formErrors.errorLabel &&
+    !formErrors.errorQuantity &&
+    !formErrors.errorDescription
+  ) {
+    if (props.currentItem) {
+      editItem(newItem);
+    } else {
+      addTreeNode(newItem, "/section/");
+    }
+  }
+}
+
+function validateForm(): void {
+  formErrors.errorLabel = validateLabel(
+    String(addForm.label),
+    getRootItemPath()
+  );
   let newItem: IItem = {
     label: addForm.label,
     icon: addForm.icon,
     favorites: false,
   };
-
-  if (checkDialogType()) {
-    formErrors.errorQuantity = checkQuantity(addForm.quantity);
-    formErrors.errorDescription = checkName(addForm.description, 264);
-    newItem = {
-      ...newItem,
-      quantity: addForm.quantity,
-      tags: addForm.tags,
-      description: addForm.description,
-    };
-    if (
-      !formErrors.errorLabel &&
-      !formErrors.errorQuantity &&
-      !formErrors.errorDescription
-    ) {
-      addItem(newItem, "/item/");
-      addForm.quantity = 0;
-      addForm.tags = [];
-      addForm.description = "";
-    }
+  if (isItem()) {
+    modifyItem(newItem);
   } else {
-    newItem = { ...newItem, items: [] };
-    if (!formErrors.errorLabel) {
-      addItem(newItem, "/section/");
-    }
+    modifySection(newItem);
   }
 }
 
-function addTag() {
-  allTagsStore.addTag(newTag.value);
-  formErrors.errorTag = checkTag(newTag.value);
+function addTag(): void {
+  formErrors.errorTag = validateTag(newTagName.value);
 
   if (!formErrors.errorTag) {
-    addForm.tags.push(newTag.value);
-    newTag.value = "";
+    addForm.tags.push({ name: newTagName.value, favorite: false });
+    newTagName.value = "";
   }
 }
 
-function removeTag(tagForRemove: string) {
-  allTagsStore.removeTag(tagForRemove);
-  addForm.tags = addForm.tags.filter((tag) => tag !== tagForRemove);
+function removeTag(tagForRemove: ITag): void {
+  addForm.tags = addForm.tags.filter((tag) => tag.name !== tagForRemove.name);
   formErrors.errorTag = "";
 }
 
-function addIcon(e: FileUploadSelectEvent) {
+function addIcon(e: FileUploadSelectEvent): void {
   addForm.icon = e.files[0].objectURL;
 }
-function removeIcon() {
+function removeIcon(): void {
   addForm.icon = "";
 }
+
+onMounted(() => {
```

Comment:
replace with one watcher

Reply:
We don`t need onUpdated now. Only onMounted left